syntax = "proto3";

package price;

import "models/util/decimal_value.proto";
import "models/util/local_timestamp.proto";
import "models/util/uuid.proto";

import "models/security/security.proto";

option java_multiple_files = true;
option java_package = "common.model.ledger-models-protos";
option java_outer_classname = "PriceProtos";

message PriceProto {
  string object_class = 1;
  string version = 2;
  //Primary Key
  util.UUIDProto uuid = 5;
  util.LocalTimestampProto as_of = 6;
  bool is_link = 7;

  util.DecimalValueProto price = 10;
  security.SecurityProto security = 11;

}
